### ISSUE TITLE
混合训练准确率下降

### ISSUE THREAD

**Issue by howardatri** [xiaomi-research/colar#13]
Hi @AlbertTan404  !
I've successfully downloaded the provided colar-best.ckpt checkpoint and run the evaluation.The results are as follows:
<img width="711" height="177" alt="Image" src="https://github.com/user-attachments/assets/0ef19eea-e0f5-4e9d-a1b5-368b1319d842" />

Comparing this to the Colar-5 metric in the paper, this result seems normal.

Subsequently, building upon this colar-best.ckpt, I attempted mixed-format training (or multi-format alignment). The original Colar output format is step -> answer. For each sample in train.json, while keeping the question unchanged, I added two new variants: answer -> step and answer only. The formats are:
[
三个变体语义定义：
- step_answer：cot + `<think>` + answer
- answer_step：answer + `<think>` + cot
- answer_only：answer（严格不出现推理 token）
]
I also rewrote the prompts for these three versions and registered the new separator <think>.

Training Configuration:
[
OMP_NUM_THREADS=1 PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True python run.py --model=colar --dataset=qsa --devices=0 --project_root=/root/autodl-tmp/workspace/colar --artifact_root=/root/autodl-tmp/workspace --model_root=/root/autodl-tmp/workspace/models --dataset_root=/root/autodl-tmp/workspace/data/text_reasoning --load_ckpt_path=/root/autodl-tmp/workspace/ckpt/logs/colar/qsa-gsm/colar-final/checkpoints/colar_best.ckpt --do_test --log_suffix=aug433_promptfix_v2_bs8x4 dataset_name=gsm enable_mixed_train=True batch_size=8 val_batch_size=8 accumulate_grad_batches=4 max_epochs=12 num_workers=4 > /root/autodl-tmp/workspace/output/aug433_promptfix_v2_run.log 2>&1
]
Note on Batch Size: Since batch_size=32 resulted in an OOM error, I used batch_size=8 with accumulate_grad_batches=4 to effectively simulate a batch size of 32.

The results were as follows:
[
Test results: defaultdict(<class 'list'>, {'monitor': [0.0803639143705368, 0.07126610726118088, 0.06823351234197617, 0.07354056090116501, 0.07278241217136383], 'test/acc': [0.0803639143705368, 0.07126610726118088, 0.06823351234197617, 0.07354056090116501, 0.07278241217136383], 'test/n_latent_forward': [39.329795837402344, 38.274452209472656, 40.071266174316406, 39.56937026977539, 39.01819610595703], 'test/n_latent_forward_on_acc': [14.899672508239746, 14.629264831542969, 14.66464614868164, 16.517562866210938, 12.629264831542969], 'test/output_length': [4.0644426345825195, 4.033358573913574, 4.018195629119873, 3.980288028717041, 4.078847408294678]})
Test statistics with 5 replications: {'monitor': (0.07323730140924453, 0.003506395940529609), 'test/acc': (0.07323730140924453, 0.003506395940529609), 'test/n_latent_forward': (39.25261611938477, 0.5242016523301881), 'test/n_latent_forward_on_acc': (14.668082237243652, 1.0829781426128342), 'test/output_length': (4.0350264549255375, 0.030535620977975873)}
]
The accuracy was only 0.07, while latent_forward skyrocketed to 39.

Question
Could you help me understand why this might be happening?
Is this due to insufficient epochs or the effective batch size being too small, leading to non-convergence?
Or, could the designed tokens be interfering with the original separator '###'?
Any insights would be greatly appreciated. Thanks!

**Comment by AlbertTan404** [comment]
Hi @howardatri, sorry for the late response since I was on a vacation.

First, I'm a little bit confused on the motivation of mixed-format training: why the model should generate answers before reasoning-steps?

Based on your information, I think the detoriated performance is simply caused by **limited model capacity** to learn such uncertain reasoning formats, without an explicit prompt to instruct the model how to perform.

If you would insist on mixed-reasoning format, I suggest a simpler variant by training the two formats together:
- Prompt + cot + answer
- Prompt + "Directly answer the question.\n" + answer
And see whether it converges.

Regarding the separator: the "###" works well and should not be changed.
